### DISCUSSION TITLE
packge.json - Allow devDependencies to override dependencies when running development version

### DISCUSSION THREAD

**Discussion by william-t-sutton** [npm/feedback#775] [General]
``Problem:

I have a few packages I wish to run locally when running my code in a development environment, and their respective deployed version when running in production. Currently, I am manually installing each local dependency when pulling a new branch, then reinstalling the deployed packages when I push my changes to my remote repo and I wish to remove the need for this process.

Potential solution:

Configuration something like the following:

```
"dependencies": {
    "@me/pack1": "^1.0.1"
    "@me/pack2": "^1.0.2"
},
"devDependencies": {
    "@me/pack1": "file:path/to/pack1"
    "@me/pack2": "file:path/to/pack2"
}
```

Is there existing functionality to achieve what I want or is there a better way to do this?

**Comment by ljharb** [comment]
You can use the npm: protocol, or you can use “overrides”.

**Reply by william-t-sutton** [reply]
I only just found this issue again. Would you be able to elaborate on what you mean?

**Reply by ljharb** [reply]
npm since v8.3 has supported "overrides" which allow you to override one dependency with another (in an app, not a published package)

https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides
